### PR TITLE
[discover] Do not queue machete repository update in project resolver…

### LINKED ISSUE #667: Autodiscover happens prematurely
Open a project with no .git/machete file. Notification shows up before Git Machete tab is even opened:

image

This might be confusing for newcomers.

I remember we've already resolved this issue... possibly it only manifests itself on 2020.3? Or a regression happened :c

### PR DESCRIPTION
**Pull request by mkondratek**
… - fixes premature autodiscover (closes #667)

### CODE REVIEW

**Review by micpiotrowski — APPROVED** [review]

File: frontendExternalSystem/src/main/java/com/virtuslab/gitmachete/frontend/externalsystem/project/MacheteProjectResolver.java [hunk 1]
```diff
@@ -28,16 +24,6 @@
       @Nullable MacheteExecutionSettings settings,
       ExternalSystemTaskNotificationListener listener)
       throws ExternalSystemException, IllegalArgumentException, IllegalStateException {
-
-    var graphTableProvider = Option.of(settings)
```
Comment (PawelLipski) [inline frontendExternalSystem/src/main/java/com/virtuslab/gitmachete/frontend/externalsystem/project/MacheteProjectResolver.java:32]: Wow okay... any idea what was the original purpose of this piece? As in, why it used to be needed back then but redundant/erroneous now?...

Comment (PawelLipski) [reply frontendExternalSystem/src/main/java/com/virtuslab/gitmachete/frontend/externalsystem/project/MacheteProjectResolver.java:32]: The point being, whether by removing this stuff we don't break sth else

**Comment by PawelLipski** [comment]
Hmm I'm getting the following error on a repo where there was NO .git/machete file:

image

But other than that, all happened as expected (autodiscover only when the tab opened, with a notification popping up)

File: frontendExternalSystem/src/main/java/com/virtuslab/gitmachete/frontend/externalsystem/project/MacheteProjectResolver.java (hunk 1 above)
Comment (mkondratek) [reply frontendExternalSystem/src/main/java/com/virtuslab/gitmachete/frontend/externalsystem/project/MacheteProjectResolver.java:32]: Now we have got this part of the code

```
addAncestorListener(new AncestorListenerAdapter() {
      @Override
      public void ancestorAdded(AncestorEvent event) {
        var gitRepository = selectedGitRepositoryProvider.getSelectedGitRepository().getOrNull();
        if (gitRepository != null) {
          var macheteFilePath = getMacheteFilePath(gitRepository);
          Runnable queueDiscoverOperation = () -> graphTable.queueDiscover(macheteFilePath, () -> {});
          var rediscoverSuggester = new RediscoverSuggester(gitRepository, queueDiscoverOperation);
          graphTable.queueRepositoryUpdateAndModelRefresh(() -> rediscoverSuggester.perform());
        }
      }
    });
```

https://github.com/VirtusLab/git-machete-intellij-plugin/blob/59f1e2000aa390475f2a50005f2acf064bb0f745/frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMachetePanel.java#L60

The reason is that we want to queue repository (first) update at the right moment. Resolver called it at the very beginning of the IDE run. In the meantime came the above part which calls the update later, on Git Machete tab opening.

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**